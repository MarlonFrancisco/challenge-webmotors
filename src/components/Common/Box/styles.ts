import styled from "styled-components";

export const Container = styled.div`
  background: #fff;
  padding: 10px 20px;
`;
