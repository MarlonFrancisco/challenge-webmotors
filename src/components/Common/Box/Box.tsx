import React, { CSSProperties } from "react";
import { Container } from "./styles";

export type BoxProps = {
  style?: CSSProperties;
};

const Box: React.FC<BoxProps> = ({ children, ...rest }) => {
  return <Container {...rest}>{children}</Container>;
};

export default Box;
