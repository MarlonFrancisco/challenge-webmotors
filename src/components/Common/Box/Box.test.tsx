import React from "react";
import { render } from "@testing-library/react";
import Box, { BoxProps } from "./Box";

const Wrapper = (props: BoxProps) => <Box {...props} />;

describe("Box", () => {
  let props: BoxProps & { children: React.ReactNode };

  beforeEach(() => {
    props = {
      children: <p>BoxComponent</p>,
    };
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render props", () => {
    const { getByText } = render(<Wrapper {...props} />);
    expect(getByText("BoxComponent").innerHTML).toEqual("BoxComponent");
  });
});
