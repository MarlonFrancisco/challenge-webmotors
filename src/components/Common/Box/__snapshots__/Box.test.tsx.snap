// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render without an error 1`] = `
<div>
  <div
    class="sc-bdnylx gQvYhD"
  >
    <p>
      BoxComponent
    </p>
  </div>
</div>
`;
